howso/mda_weight.amlg

```diff
@@ -0,0 +1,436 @@
+(null
+
+	;helper method for CalculateFeatureResiduals to calculate 'superfull' accurate residuals
+	#!RunAccurateResiduals
+	(seq
+		;keep a copy of all originally specified features
+		(assign (assoc case_features features))
+
+		;if we only collected residuals for this one feature, set the residuals features list to have only target_residual_feature
+		(if target_residual_feature
+			(assign (assoc features (list target_residual_feature) ))
+		)
+
+		;create a map of feature -> flag, set to true if the feature has any null values
+		;that way there's no need to explicitly filter nulls out when querying for a feature that no nulls
+		(declare (assoc
+			feature_may_have_nulls_map
+				(map
+					(lambda (!= (false) (get !featureNullRatiosMap (list (current_index 1) "has_nulls")) ) )
+					(zip
+						(if target_residual_feature
+							(list target_residual_feature)
+							features
+						)
+					)
+				)
+			series_id_features (if !tsTimeFeature (get !tsModelFeaturesMap "series_id_features") )
+		))
+
+		;iterate over each case and accumulate residuals for all the feature(s)
+		#!AccumulateAccurateFeatureResiduals
+		(assign (assoc
```

Comment:
To match the other FeatureResiduals methods, this code should support having a separate lists of `features`: the features to compute the residuals for, and `context_features`: the features to use as contexts (which can overlap and probably will usually overlap)